app/Main.hs

```diff
@@ -61,11 +64,19 @@ main = do
 
     runCardanoApplicationWithFeatures Development cardanoFeatures (cardanoApplication nodeLayer)
 
-initializeAllFeatures :: CLIArguments -> CardanoConfiguration -> CardanoEnvironment -> IO ([CardanoFeature], NodeLayer)
-initializeAllFeatures (CLIArguments logCli nodeCli) cardanoConfiguration cardanoEnvironment = do
+initializeAllFeatures :: CLIArguments -> PartialCardanoConfiguration -> CardanoEnvironment -> IO ([CardanoFeature], NodeLayer)
+initializeAllFeatures (CLIArguments logCli nodeCli) partialConfig cardanoEnvironment = do
+    finalConfig <- either (throwIO . ConfigurationError) pure $
+          finaliseCardanoConfiguration $
+          -- Here we perform merging of layers of configuration, but for now, only in a trivial way,
+          -- just for Cardano.Shell.Constants.Types.Genesis.
+          -- We expect this process to become generic at some point.
+          partialConfig { pccCore =
```

Comment:
I think this is still a bit confusing - I think it would be worth pulling this out into a function with more comments

Reply:
Done!